segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/DebugStreamSegmentContainerTests.java

```diff
@@ -322,18 +323,20 @@ public void testBackUpMetadataAndAttributeSegments() {
     public void testDataRecoveryContainerLevel() throws Exception {
         int attributesUpdatesPerSegment = 50;
         int segmentsCount = 10;
-        final UUID attributeReplace = UUID.randomUUID();
+        final UUID attributeReplace = new UUID(CORE_ATTRIBUTE_ID_PREFIX, RANDOM.nextInt(10));
         final long expectedAttributeValue = attributesUpdatesPerSegment;
         int containerId = 0;
         int containerCount = 1;
         int maxDataSize = 1024 * 1024; // 1MB
 
+        StorageFactory storageFactory = new InMemoryStorageFactory(executorService());
+
```

Comment:
Created a storage factory beforehand and use it for recovery. This will make the text more clear. Previously we used the same context (`storageFactory`, `readIndexFactory`, `attributeIndexFactory` etc.) except `DurableLog` before and after the recovery.